iModelCore/Units/tests/NonPublished/FormattingTests.cpp

```diff
@@ -1541,4 +1556,519 @@ TEST_F(FormattingTestFixture, FormatsUsingDefaultNumericFormatSpecWhenItsNotSet)
+    formatRatioTest("0.667:1", 0.6667, v_h, NtoOne, v_h);
+    }
+
+    // v:h -> h:v | N to one
+    {
+    formatRatioTest("1:1", 1.0, v_h, NtoOne, h_v);
+    formatRatioTest("0.5:1", 2.0, v_h, NtoOne, h_v);
+    formatRatioTest("2:1", 0.5, v_h, NtoOne, h_v);
+    formatRatioTest("3.003:1", 0.333, v_h, NtoOne, h_v);
+    formatRatioTest("3:1", 0.3333, v_h, NtoOne, h_v);
+    formatRatioTest("3.5:1", 0.2857, v_h, NtoOne, h_v);
+    formatRatioTest("4:1", 0.25, v_h, NtoOne, h_v);
+    formatRatioTest("1.5:1", 0.6667, v_h, NtoOne, h_v);
+    }
+
+    {
+    // decimal precision
+    // formatRatioTest("1:0", 3, v_h, oneToN, v_h, DecimalPrecision::Precision0);
+    formatRatioTest("1:0.3", 3, v_h, oneToN, v_h, DecimalPrecision::Precision1);
+    formatRatioTest("1:0.33", 3, v_h, oneToN, v_h, DecimalPrecision::Precision2);
+
+    formatRatioTest("1:2500", 0.0004, v_h, oneToN, v_h, DecimalPrecision::Precision3);
+
+    formatRatioTest("1:0", 0.0004, v_h, oneToN, h_v, DecimalPrecision::Precision3);
+    formatRatioTest("1:0.0004", 0.0004, v_h, oneToN, h_v, DecimalPrecision::Precision4);
+    formatRatioTest("1:-0.0004", -0.0004, v_h, oneToN, h_v, DecimalPrecision::Precision4);
+    }
+
+    // v:h -> v:h | ValueBased
+    // if the value is smaller than 1, its 1 to N. else, N to 1
+    {
+    formatRatioTest("1:1", 1.0, v_h, valueBased, v_h);
+    formatRatioTest("2:1", 2.0, v_h, valueBased, v_h);
+    formatRatioTest("1:2", 0.5, v_h, valueBased, v_h);
+    formatRatioTest("1:3.003", 0.333, v_h, valueBased, v_h);
+    formatRatioTest("3.333:1", 3.3333, v_h, valueBased, v_h);
+    formatRatioTest("1:3.5", 0.2857, v_h, valueBased, v_h);
+    formatRatioTest("3.5:1", 3.5, v_h, valueBased, v_h);
+    formatRatioTest("1:4", 0.25, v_h, valueBased, v_h);
+    formatRatioTest("4:1", 4.0, v_h, valueBased, v_h);
+    }   
+
+    // v:h -> v:h | UseGreatestCommonDivisor
+    {
+    formatRatioTest("1:1", 1.0, v_h, useGreatestCommonDivisor, v_h);
+    formatRatioTest("2:1", 2.0, v_h, useGreatestCommonDivisor, v_h);
+    formatRatioTest("1:2", 0.5, v_h, useGreatestCommonDivisor, v_h);
+    formatRatioTest("333:1000", 0.333, v_h, useGreatestCommonDivisor, v_h);
+    formatRatioTest("333:1000", 0.3333, v_h, useGreatestCommonDivisor, v_h);
+    formatRatioTest("143:500", 0.2857, v_h, useGreatestCommonDivisor, v_h);
+    formatRatioTest("1:4", 0.25, v_h, useGreatestCommonDivisor, v_h);
+    formatRatioTest("667:1000", 0.6667, v_h, useGreatestCommonDivisor, v_h);
+    }
+
+    // negative values
+    {
+    formatRatioTest("-1:1", -1.0, v_h, NtoOne, v_h);
+    formatRatioTest("1:-1", -1.0, v_h, valueBased, v_h);
+
+    formatRatioTest("-0.5:1", -0.5, v_h, NtoOne, v_h);
+    formatRatioTest("1:-2", -0.5, v_h, valueBased, v_h);
+
+    formatRatioTest("-0.5:1", -2, v_h, NtoOne, h_v);
+    formatRatioTest("1:-2", -2, v_h, valueBased, h_v);
+    }
+
+    // really large/small numbers
+    {
+    formatRatioTest("0:1", 0.00000001, v_h, NtoOne, v_h);

```

Comment:
I don't get this one

Reply:
the format has precision:3
input 0.00000001 gets rounded to 0, 
so the output string is 0:1